### PR TITLE
Add HTTP semantic convention stability migration guide

### PR DESCRIPTION
**Pull request by trask**
Part of #534

## Changes

Adds a "permanent" (non-blog) location for HTTP semantic convention stability migration guide.

@open-telemetry/docs-approvers I copied this from https://opentelemetry.io/blog/2023/http-conventions-declared-stable/#migration-plan and stripped out the non-markdown stuff. what do you think of rendering this as a new page under https://opentelemetry.io/docs/specs/semconv/http/?

## Merge requirement checklist

* [x] [CONTRIBUTING.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CONTRIBUTING.md) guidelines followed.
* ~[CHANGELOG.md](https://github.com/open-telemetry/semantic-conventions/blob/main/CHANGELOG.md) updated for non-trivial changes.~
* ~[schema-next.yaml](https://github.com/open-telemetry/semantic-conventions/blob/main/schema-next.yaml) updated with changes to existing conventions.~

### CODE REVIEW

**Review by reyang — APPROVED** [review]

**Review by AlexanderWert — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

File: docs/http/migration-guide.md [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+Due to the significant number of modifications and the extensive user base
+affected by them, existing HTTP instrumentations published by
+OpenTelemetry are required to implement a migration plan that will assist users in
+transitioning to the stable HTTP semantic conventions.
+
+Specifically, when existing HTTP instrumentations published by OpenTelemetry are
+updated to the stable HTTP semantic conventions, they:
+
+- Need to introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN` in
```
Comment (TylerHelmuth) [inline docs/http/migration-guide.md:11]: Is `Need` the proper "level" here?  I thought we used `SHOULD` for the env var.

File: docs/http/migration-guide.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+- May drop the environment variable in their next major version and emit only
+  the stable HTTP and networking conventions.
+
+## Summary of changes
+
+This section summarizes the changes made to the HTTP semantic conventions
+from
+[v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md)
+to
+[v1.23.0 (stable)](README.md).
```
Comment (lmolkova) [inline docs/http/migration-guide.md:33]:
```suggestion
[v1.23.1 (stable)](README.md).
```

File: docs/http/migration-guide.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+| `http.scheme` &rarr; `url.scheme` | Now factors in [X-Forwarded-Proto][], [Forwarded#proto][] headers |
+| `http.client_ip` &rarr; `client.address` | If `http.client_ip` was unknown (i.e., no [X-Forwarded-For][], [Forwarded#for][] headers), then `net.sock.peer.addr` &rarr; `client.address`; now must be provided to sampler |
+| `net.host.name` &rarr; `server.address` | Now based only on [Host][Host header], [:authority][HTTP/2 authority], [X-Forwarded-Host][], [Forwarded#host][] headers |
+| `net.host.port` &rarr; `server.port` | Now based only on [Host][Host header], [:authority][HTTP/2 authority], [X-Forwarded-Host][X-Forwarded-Host], [Forwarded#host][] headers |
+<!-- prettier-ignore-end -->
+
+References:
+
+- [HTTP server span attributes v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#http-server)
+- [HTTP server span attributes v1.23.0 (stable)](http-spans.md#http-server)
```
Comment (lmolkova) [inline docs/http/migration-guide.md:94]:
```suggestion
- [HTTP server span attributes v1.23.1 (stable)](http-spans.md#http-server)
```

File: docs/http/migration-guide.md [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+| `net.sock.peer.port` &rarr; `network.peer.port` | Now captured even if same as `server.port` |
+| `net.sock.peer.name` | Removed |
+| New: `http.request.method_original` | Only captured when `http.request.method` is `_OTHER` |
+| New: `error.type` |  |
+<!-- prettier-ignore-end -->
+
+References:
+
+- [Common attributes v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#common-attributes)
+- [Common attributes v1.23.0 (stable)](http-spans.md#common-attributes)
```
Comment (lmolkova) [inline docs/http/migration-guide.md:60]:
```suggestion
- [Common attributes v1.23.1 (stable)](http-spans.md#common-attributes)
```

File: docs/http/migration-guide.md [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+| `http.url` &rarr; `url.full` | |
+| `http.resend_count` &rarr; `http.request.resend_count` | |
+| `net.peer.name` &rarr; `server.address` | |
+| `net.peer.port` &rarr; `server.port` | Now captured even when same as default port for scheme |
+<!-- prettier-ignore-end -->
+
+References:
+
+- [HTTP client span attributes v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#http-client)
+- [HTTP client span attributes v1.23.0 (stable)](http-spans.md#http-client)
```
Comment (lmolkova) [inline docs/http/migration-guide.md:76]:
```suggestion
- [HTTP client span attributes v1.23.1 (stable)](http-spans.md#http-client)
```

File: docs/http/migration-guide.md [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+
+The `{http.method}` portion of span names is replace by `HTTP` when
+`{http.method}` is `_OTHER`.
+
+See note if [migrating from `<= v1.17.0`](#migrating-from--v1170).
+
+References:
+
+- [HTTP client and server span names v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/trace/semantic_conventions/http.md#name)
+- [HTTP client and server span names v1.23.0 (stable)](http-spans.md#name)
```
Comment (lmolkova) [inline docs/http/migration-guide.md:106]:
```suggestion
- [HTTP client and server span names v1.23.1 (stable)](http-spans.md#name)
```

File: docs/http/migration-guide.md [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+| `net.host.name` → `server.address` | &bullet; Recommended → Opt-In (due to high-cardinality vulnerability since based on HTTP headers)<br>&bullet; Now based only on [`Host` span][Host header], [`:authority` span][HTTP/2 authority], [`X-Forwarded-Host` span][X-Forwarded-Host], [`Forwarded#host` span][Forwarded#host] headers |
+| `net.host.port` → `server.port` | &bullet; Recommended → Opt-In (due to high-cardinality vulnerability since based on HTTP headers)<br>&bullet; Now based only on [`Host` span][Host header], [`:authority` span][HTTP/2 authority], [`X-Forwarded-Host` span][X-Forwarded-Host], [`Forwarded#host` span][Forwarded#host] headers |
+| New: `error.type` |  |
+<!-- prettier-ignore-end -->
+
+References:
+
+- [Metric `http.server.duration` v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/metrics/semantic_conventions/http-metrics.md#metric-httpserverduration)
+- [Metric `http.server.request.duration` v1.23.0 (stable)](http-metrics.md#metric-httpserverrequestduration)
```
Comment (lmolkova) [inline docs/http/migration-guide.md:167]:
```suggestion
- [Metric `http.server.request.duration` v1.23.1 (stable)](https://github.com/open-telemetry/semantic-conventions/blob/v1.23.1/docs/http/http-metrics.md#metric-httpserverrequestduration)
```

File: docs/http/migration-guide.md [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+| `net.sock.peer.addr` | Removed |
+| `net.protocol.name` &rarr; `network.protocol.name` | Recommended &rarr; Conditionally required if not `http` and `network.protocol.version` is set |
+| `net.protocol.version` &rarr; `network.protocol.version` | Examples fixed: `2.0` &rarr; `2` and `3.0` &rarr; `3`; see note if [migrating from `<= v1.19.0`](#migrating-from--v1190) |
+| New: `error.type` |  |
+<!-- prettier-ignore-end -->
+
+References:
+
+- [Metric `http.client.duration` v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/metrics/semantic_conventions/http-metrics.md#metric-httpclientduration)
+- [Metric `http.client.request.duration` v1.23.0 (stable)](http-metrics.md#metric-httpclientrequestduration)
```
Comment (lmolkova) [inline docs/http/migration-guide.md:136]:
```suggestion
- [Metric `http.client.request.duration` v1.23.1 (stable)](https://github.com/open-telemetry/semantic-conventions/blob/v1.23.1/docs/http/http-metrics.md#metric-httpserverrequestduration)
```

File: docs/http/migration-guide.md [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+| New: `error.type` |  |
+<!-- prettier-ignore-end -->
+
+References:
+
+- [Metric `http.server.duration` v1.20.0](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/metrics/semantic_conventions/http-metrics.md#metric-httpserverduration)
+- [Metric `http.server.request.duration` v1.23.0 (stable)](http-metrics.md#metric-httpserverrequestduration)
+
+## Migrating from a version earlier than v1.20.0?
+
```
Comment (lmolkova) [inline docs/http/migration-guide.md:170]: I think it would be helpful to add a quick note that below points are in additional to things listed above

**Review by lmolkova — APPROVED** [review]

File: docs/http/migration-guide.md (hunk 1 above)
Comment (cartermp) [reply docs/http/migration-guide.md:11]: Yep, much as I would like this to be a requirement, I was also under the impression that it was optional.

File: docs/http/migration-guide.md [hunk 10]
```diff
@@ -0,0 +1,212 @@
+# HTTP semantic convention stability migration guide
+
+Due to the significant number of modifications and the extensive user base
+affected by them, existing HTTP instrumentations published by
+OpenTelemetry are required to implement a migration plan that will assist users in
```
Comment (cartermp) [inline docs/http/migration-guide.md:5]: Might be good to spell out that this period is intended to be 6 months, or some timeline here. That's only inferable when you look later on and see the need to maintain the old values for that period of time.

File: docs/http/migration-guide.md (hunk 1 above)
Comment (trask) [reply docs/http/migration-guide.md:11]:
```suggestion
- SHOULD introduce an environment variable `OTEL_SEMCONV_STABILITY_OPT_IN` in
```

**Review by TylerHelmuth — APPROVED** [review]

File: docs/http/migration-guide.md (hunk 10 above)
Comment (trask) [reply docs/http/migration-guide.md:5]: can you clarify? do you mean something other than

> - Need to maintain (security patching at a minimum) their existing major version
  for at least six months after it starts emitting both sets of conventions.

File: .markdown_link_check_config.json [hunk 11]
```diff
@@ -10,7 +10,7 @@
       "replacement": "{{BASEURL}}/"
     },
     {
-      "pattern": "^https://github.com/open-telemetry/semantic-conventions/(blob|tree)/[^/]+/docs/",
+      "pattern": "^https://github.com/open-telemetry/semantic-conventions/(blob|tree)/[^v][^/]*/docs/",
       "replacement": "LINK-CHECK-ERROR-USE-LOCAL-PATH-TO-DOC-PAGE-NOT-EXTERNAL-URL/"
```
Comment (trask) [inline .markdown_link_check_config.json:14]: @open-telemetry/docs-maintainers I added an exclusion here for referencing tagged content (`v*`), is this ok?

**Review by joaopgrassi — APPROVED** [review]

File: .markdown_link_check_config.json (hunk 11 above)
Comment (trask) [reply .markdown_link_check_config.json:14]: resolving for now so this can be merged, and we can figure this out later

**Pull request merged**